receiver/awscontainerinsightreceiver/internal/k8swindows/k8swindows.go

```diff
@@ -0,0 +1,106 @@
+package k8swindows
+
+import (
+	"context"
+	"errors"
+	"os"
+
+	ci "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/containerinsight"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/cadvisor/extractors"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/host"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/stores"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+type K8sWindows struct {
+	cancel          context.CancelFunc
+	logger          *zap.Logger
+	nodeName        string `toml:"node_name"`
+	k8sDecorator    stores.K8sDecorator
+	summaryProvider *kubeletSummaryProvider
+	hostInfo        host.Info
+}
+
+func New(logger *zap.Logger, decorator *stores.K8sDecorator, hostInfo host.Info) (*K8sWindows, error) {
+	nodeName := os.Getenv("HOST_NAME")
+	if nodeName == "" {
+		return nil, errors.New("missing environment variable HOST_NAME. Please check your deployment YAML config")
+	}
+	k8sSummaryProvider, err := new(logger, hostInfo)
+	if err != nil {
+		logger.Error("failed to initialize kubelet summary provider, ", zap.Error(err))
+		return nil, err
+	}
+	return &K8sWindows{
+		logger:          logger,
+		nodeName:        nodeName,
+		k8sDecorator:    *decorator,
+		summaryProvider: k8sSummaryProvider,
+		hostInfo:        hostInfo,
+	}, nil
+}
+
+func (k *K8sWindows) GetMetrics() []pmetric.Metrics {
+	k.logger.Debug("D! called K8sWindows GetMetrics")
+	var result []pmetric.Metrics
+
+	metrics, err := k.summaryProvider.getMetrics()
+	if err != nil {
+		k.logger.Error("error getting metrics from kubelet summary provider, ", zap.Error(err))
+		return result
+	}
+	metrics = k.decorateMetrics(metrics)
+	for _, cadvisorMetric := range metrics {
```

Comment:
nit: Can the metric name in iteration be something like `k8s_summary_metric`?

Reply:
Addressed